src/presets.rs

```diff
@@ -239,7 +214,7 @@ pub fn is_safe_prime_with_rng<T: Integer + RandomMod>(rng: &mut (impl CryptoRng
         return false;
     }
 
-    is_prime_with_rng(rng, candidate) && is_prime_with_rng(rng, &candidate.wrapping_shr_vartime(1))
+    is_prime(candidate) && is_prime(&candidate.wrapping_shr_vartime(1))
```

Comment:
In theory, shouldn't it be faster to check the smaller candidate first? I tried it and found no measurable difference in performance so I guess not...

Reply:
It'll make a small difference in Lucas test (one less Lucas sequence propagation), and only a small difference in MR test if dividing the candidate by two leads to one less window during exponentiation. And the MR test comes first. So I theoretically it can be an improvement, but a tiny one.